h/views/accounts.py

```diff
@@ -492,19 +493,28 @@ def _template_data(self):
         log_in_with_orcid = feature_service.enabled(
             "log_in_with_orcid", self.request.user
         )
-        orcid = orcid_url = None
+        orcid_id = orcid_url = None
         if log_in_with_orcid:
-            orcid_client = self.request.find_service(ORCIDClientService)
-            orcid_identity = orcid_client.get_identity(self.request.user)
-            orcid = orcid_identity.provider_unique_id if orcid_identity else None
-            orcid_url = orcid_client.orcid_url(orcid)
+            orcid_host = self.request.registry.settings["orcid_host"]
+            oidc_service = self.request.find_service(OIDCService)
+            orcid_identity = oidc_service.get_identity(
+                self.request.user, IdentityProvider.ORCID
+            )
+            orcid_id = orcid_identity.provider_unique_id if orcid_identity else None
+
+            if orcid_id:
+                # The URL to the user's public ORCID profile page
+                # (for example: https://orcid.org/0000-0002-6373-1308).
+                orcid_url = urlunparse(urlparse(orcid_host)._replace(path=orcid_id))
+            else:
+                pass  # pragma: no cover
```

Comment:
This code was previously calling `ORCIDClientService.orcid_url(orcid)` (where `orcid` was the user's ORCID iD) to get the URL to the user's ORCID profile page. `ORCIDClientService` has now been replaced with `OIDCService` and these profile pages aren't an OIDC thing so the new service doesn't have this method. So had the view code format the profile page URL itself as well.